src/Controls/src/BindingSourceGen/DiagnosticsFactory.cs

```diff
@@ -81,4 +81,37 @@ public static DiagnosticInfo LambdaResultCannotBeResolved(Location location)
 				defaultSeverity: DiagnosticSeverity.Error,
 				isEnabledByDefault: true),
 			location);
+
+	public static DiagnosticInfo UnaccessibleTypeUsedAsLambdaParameter(Location location)
+		=> new DiagnosticInfo(
+			new DiagnosticDescriptor(
+				id: "BSG0007",
+				title: "Unaccessible type used as lambda parameter",
+				messageFormat: "The lambda parameter type has to be declared as public, internal or protected internal.",
```

Comment:
This is separate to this PR, but do we need to file an issue to localize this text?

Reply:
These will be visible to users during compile time, so I think, eventually, yes. We are actively expanding the capabilities of SG bindings, so the diagnostics list will likely change. Should we start localizing them now? If so, what is the process?

Reply:
I was looking at the other source generator, and it has `.resx` but nothing translated...

https://github.com/dotnet/maui/blob/main/src/Controls/src/SourceGen/MauiGResources.resx

An example of how this works is on the history of:
* https://github.com/dotnet/maui/commits/main/src/Controls/src/Build.Tasks/xlf

I think someone needs to localize the other source generator as well (but not your team), so I'll ask about it.

Reply:
Yeah I need to open a ticket for tracking localization effort.